wsi_deid/web_client/views/ItemList.js

```diff
@@ -171,6 +181,110 @@ wrap(ItemListWidget, 'render', function (render) {
         return result;
     };
 
+    const generateStringFromPattern = (pattern) => {
+        const letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+        const randomLetter = () => letters.charAt(Math.floor(Math.random() * letters.length));
+        const randomNumber = () => Math.floor(Math.random() * 10);
+        const result = pattern.split('');
+        result.forEach((char, index) => {
+            let newChar = char;
+            if (char === '#') {
+                newChar = randomNumber();
+            } else if (char === '@') {
+                newChar = randomLetter();
+            }
+            result[index] = newChar;
+        });
+        return result.join('');
+    };
+
+    const refileCheckedItems = () => {
+        const togetherSelect = $('.g-refile-select-togetherness');
+        const newOrExistingSelect = $('.g-refile-select-new-or-existing');
+        const fileTogether = togetherSelect.find(':selected').val() === 'together';
+        const useExistingToken = newOrExistingSelect.find(':selected').val() === 'existing';
+        const tokenPattern = this._wsi_deid_settings['new_token_pattern'];
```

Comment:
We should set a default for the new_token_pattern.  If it isn't in the config file, the code throws and error.

Reply:
See e81fcae